### PR TITLE
gegl, babl: fixes for PPC and <10.6 Intel

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

This PR addresses a few bugs:

1. Invalid option in meson.build used for PPC (cannot be boolean): https://trac.macports.org/ticket/66233
2. No OpenCL on <10.6 Intel and any PPC (has been thrown away prior to release or not implemented at all).
3. Blacklisting GCC – no reason for that, GCC works fine. Moreover, on PPC it is the only compiler.
4. Vapigen is broken on PPC, related default variants should not be used there.

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.6
Xcode 3.2

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: graphics/gegl/Portfile [hunk 1]
```diff
@@ -73,6 +73,15 @@ patchfiles-append   patch-gegl-32bit-host-statistics.diff
 # https://trac.macports.org/ticket/66667
 patchfiles-append   patch-fix-raw-load.c.diff
 
+# https://trac.macports.org/ticket/66233
+if {[string match *gcc* ${configure.compiler}]} {
+    patchfiles-append patch-use-gcc.diff
+}
+
+if {${os.major} < 10 || (${os.major} == 10 && ${build_arch} eq "ppc")} {
```
Comment (reneeotten) [inline graphics/gegl/Portfile:81]: missing check for os.platform

**Review by reneeotten — CHANGES_REQUESTED** [review]
@barracuda156 : one comment but these changes look nicely restricted to older systems. There has been sufficient time for maintainers to review this so I'm fine with merging after the few issues are addressed.

**Comment by reneeotten** [comment]
@barracuda156 is this only fixing build failure and thus doesn't require a "revision" increase? Other than that I'll merge; this has been around in the queue (too) long (enough) by now...

File: graphics/gegl/Portfile (hunk 1 above)
Comment (barracuda156) [reply graphics/gegl/Portfile:81]: I will it now, sorry, missed.

**Comment by barracuda156** [comment]
> @barracuda156 is this only fixing build failure and thus doesn't require a "revision" increase? Other than that I'll merge; this has been around in the queue (too) long (enough) by now...

@reneeotten Will finish this today. Found an improvement over what we have here: `gegl` allows to use `libsdl1` as an option, we can use it for old systems.

**Comment by barracuda156** [comment]
@reneeotten Okay, once CI pass, this should be ready.

**Comment by barracuda156** [comment]
@reneeotten Something left to fix or good to merge?

**Pull request merged**